### PR TITLE
Add expert partitioner specs

### PR DESCRIPTION
**Pull request by joseivanlopez**
PBI: https://trello.com/c/ZEkBVzkj/661-2-partitioner-specification-plain-partitions.

### CODE REVIEW

File: doc/user-stories/expert-partitioner.md [hunk 1]
```diff
@@ -0,0 +1,462 @@
+
+# Expert partitioner specs
+
+## When 'Hard Disks' is selected in the tree view
+  * shows a table with the columns: Device, Size, F, Enc, Type, FS Type, Label, Mount Point, Start, End
```
Comment (jreidinger) [inline doc/user-stories/expert-partitioner.md]: where is code that runs this statements?

Comment (jreidinger) [inline doc/user-stories/expert-partitioner.md]: and in fact old expert partitioner have here configurable columns, try change columns in options. I am just not sure if we want it also in new storage-ng as it is quite hidden and not much intuitive.

File: doc/user-stories/expert-partitioner.md [hunk 2]
```diff
@@ -0,0 +1,414 @@
...
+    * Format partition: true
+      * File System: XFS
+    * Do not format partition: false
+      * File System Id: 0x83 Linux
+    * Encrypt: false
+  * sets the following options for 'Mounting options'
+    * Mount partition: true
+      * Mount point: first free of /, /home, /var
+    * Do not mount partition: false
+* *(pending) and partition role is 'Swap'*
```
Comment (jreidinger) [inline doc/user-stories/expert-partitioner.md]: why pending?

File: doc/user-stories/expert-partitioner.md [hunk 3]
```diff
@@ -0,0 +1,414 @@
...
+* *(pending) and partition role is 'Raw volume'*
+  * sets the following 'Formatting options' by default
+    * Format partition: false
+    * Do not format partition: true
+      * File System Id: 0x8E Linux LVM
+    * Encrypt: false
+  * sets the following options for 'Mounting options'
+    * Mount partition: false
+    * Do not mount partition: true
+* *(pending) and 'Finish' button is selected*
```
Comment (jreidinger) [inline doc/user-stories/expert-partitioner.md]: this does not work?

File: doc/user-stories/expert-partitioner.md [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+* allows to select a 'File system'
+  * where options are: BtrFs, EXT2, EXT3, EXT4, FAT, XFS, Swap 
+* *(pending) and 'File system' is BtrFS*
+  * does not show 'Options' button
+  * shows 'Enable snaphsots'
+  * and mount point is /
+    * marks 'Enable snapshots'
+  * shows button 'Subvolume Handling'
+  * sets 'File system Id' to 0x83 Linux
+* *(pending) and 'File system' is EXT2*
```
Comment (jreidinger) [inline doc/user-stories/expert-partitioner.md]: what exactly is pending here? I thought that only some parts of ext2 is missing

File: doc/user-stories/expert-partitioner.md [hunk 5]
```diff
@@ -0,0 +1,414 @@
...
+* *(pending) when 'File system Id' is 0x8E Linux LVM, 0xFD Linux RAID, 0xA0 Hibernation, 0x00 BIOS Grub, 0x00 GPT PReP Boot*
+  * disables 'Mounting options'
+* when 'File system Id' is 0x82 Linux swap
+  * enables 'Mounting options'
+  * sets 'Mount point' to swap
+
+#### When 'Mount partition' is selected
+* allows to select 'Mount point'
+  * where options are: /, /home, /var, /opt, /boot, /srv, /tmp, /usr/local
+  * *(pending) first free value is taken by default*
```
Comment (jreidinger) [inline doc/user-stories/expert-partitioner.md]: I think it depends on role. For OS it select / and for data it use /home, not?

File: doc/user-stories/expert-partitioner.md (hunk 2 above)
Comment (joseivanlopez) [inline doc/user-stories/expert-partitioner.md]: In the new version, when you select role 'Swap', then in the next step the options "Format device" and "Mount Device" are not selected by default. Ok, I will set as pending only this two options.

File: doc/user-stories/expert-partitioner.md (hunk 3 above)
Comment (joseivanlopez) [inline doc/user-stories/expert-partitioner.md]: I don't think so. First, validations are not done and it seems that selected values are not saved, or at least they are not correctly loaded when then you edit.

File: doc/user-stories/expert-partitioner.md (hunk 4 above)
Comment (joseivanlopez) [inline doc/user-stories/expert-partitioner.md]: Yes, you are right. Missing parts are only regarding options button.

File: doc/user-stories/expert-partitioner.md [hunk 6]
```diff
@@ -0,0 +1,419 @@
...
+      * shows an error popup
+    * resizes the partition
+
+### When 'Add partition' is selected
+* and there is not space enough
+  * shows an error popup
+* and it is not possible to create more partitions (max number of primary reached)
+  * shows an error popup
+* and it is possible to create a new partition
+  * shows a wizard with 4 steps
```
Comment (aschnell) [inline doc/user-stories/expert-partitioner.md]: Up to five steps if encryption is selected.

File: doc/user-stories/expert-partitioner.md [hunk 7]
```diff
@@ -0,0 +1,419 @@
...
+* and a 'Mount point' is indicated
+  * shows the button 'Fstab options'
+
+##### When 'Do not mount partition' is selected
+* *(pending) disables 'Mount device' options*
+
+##### When 'Fstab options' is selected
+* and 'File system' is BtrFs
+  * shows a dialog with the following fields
+    * Mount in /etc/fstab by: Device name, Volume label, UUID (default), Device ID, Device path
```
Comment (aschnell) [inline doc/user-stories/expert-partitioner.md]: The default is configurable. Device ID and path disabled if not available.

File: doc/user-stories/expert-partitioner.md [hunk 8]
```diff
@@ -0,0 +1,419 @@
...
+    * Mount in /etc/fstab by: Device name, Volume label, UUID (default), Device ID, Device path.
+    * Volume label:
+    * Swap priority:
+    * Arbitrary option value:
+
+##### *(pending) When 'Subvolume handling' is selected*
+* shows a dialog with a list of all subvolumes
+* allows to add a new subvolume
+* allows to remove one subvolume from the list
+* alerts when trying to create a subvolume that does not start by /@
```
Comment (aschnell) [inline doc/user-stories/expert-partitioner.md]: The starting @ is variable (see control.xml).

File: doc/user-stories/expert-partitioner.md [hunk 9]
```diff
@@ -0,0 +1,419 @@
...
+    * *(pending) shows a button for 'Health Test (SMART)'*
+  * and tab 'Partitions' is selected
+    * shows a bar image with the disk partitions
+    * shows a table filled out with disk partitions info (same columns than before)
+    * shows the actions: 'Expert', 'Add', 'Edit', 'Move'
+      * *(pending) and action 'Expert' is selected*
+        * shows two options: 'Create New Partition Table' and 'Clone this Disk'
+          * and 'Create New Partition Table' is selected
+            * shows a popup to select a partition table type: MSDOS, GPT, DASD
+            * DASD is only showed for s390 arch
```
Comment (aschnell) [inline doc/user-stories/expert-partitioner.md]: Depends on DASD type (ECKD or FBA).

**Pull request merged**